front/src/applications/stdcm/components/StdcmLoader.tsx

```diff
@@ -1,42 +1,57 @@
-import { forwardRef } from 'react';
+import { useEffect } from 'react';
 
 import { Button } from '@osrd-project/ui-core';
 import { useTranslation } from 'react-i18next';
 
-import stdcmLoaderImg from 'assets/pictures/views/stdcm_loader.jpg';
+const LOADER_OFFSET = 32;
 
 type StdcmLoaderProps = {
   cancelStdcmRequest: () => void;
 };
 
-const StdcmLoader = forwardRef(
-  ({ cancelStdcmRequest }: StdcmLoaderProps, ref: React.Ref<HTMLDivElement>) => {
-    const { t } = useTranslation('stdcm');
-    return (
-      <div className="stdcm-loader-background">
-        <div
-          ref={ref}
-          className="stdcm-loader d-flex flex-column justify-content-center align-items-center"
-        >
-          <div className="stdcm-loader__wrapper">
-            <h1>{t('simulation.calculatingSimulation')}</h1>
-            <p>{t('simulation.averageRequestTime')}</p>
-          </div>
-          <div className="stdcm-loader__cancel-btn">
-            <Button
-              data-testid="cancel-simulation-button"
-              variant="Cancel"
-              label={t('simulation.stopCalculation')}
-              size="small"
-              onClick={cancelStdcmRequest}
-            />
-          </div>
-          <img src={stdcmLoaderImg} alt={t('simulation.pendingSimulation')} />
-          <p className="stdcm-loader__img-signature">{t('loaderImageLegend')}</p>
+const StdcmLoader = ({ cancelStdcmRequest }: StdcmLoaderProps) => {
+  const { t } = useTranslation('stdcm');
+
+  useEffect(() => {
+    // Depending on the scroll change the position of the loader between fixed and sticky
+    const handleScroll = () => {
+      const element = document.querySelector('.stdcm-loader') as HTMLElement;
```

Comment:
This is a bit fragile: React can re-render the element at any time, in which case our manual DOM changes here would be lost.

Would be nicer to add a state storing whether we're scrolled up or not, and the set a class on the loader element based on that state.

Reply:
You're right, done